ni/measurements/metadata/v1/registration.schema.json

```diff
@@ -0,0 +1,407 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Digital Thread Data Schema",
+  "description": "Schema for validating JSON files used to describe entities that should be added to the digital thread.",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "$schema": {
+      "type": "string",
+      "description": "The JSON Schema URI for validation"
+    },
+    "hardwareItems": {
+      "type": "array",
+      "description": "Collection of hardware items used to take measurements",
+      "items": {
+        "$ref": "#/definitions/HardwareItem"
+      }
+    },
+    "operators": {
+      "type": "array",
+      "description": "Collection of operators who performed test operations",
+      "items": {
+        "$ref": "#/definitions/Operator"
+      }
+    },
+    "softwareItems": {
+      "type": "array",
+      "description": "Collection of software items used to take measurements",
+      "items": {
+        "$ref": "#/definitions/SoftwareItem"
+      }
+    },
+    "testAdapters": {
+      "type": "array",
+      "description": "Collection of test adapters (fixtures, interface boards, or mechanical setups)",
+      "items": {
+        "$ref": "#/definitions/TestAdapter"
+      }
+    },
+    "tests": {
+      "type": "array",
+      "description": "Collection of test definitions",
+      "items": {
+        "$ref": "#/definitions/Test"
+      }
+    },
+    "testStations": {
+      "type": "array",
+      "description": "Collection of test stations where tests are executed",
+      "items": {
+        "$ref": "#/definitions/TestStation"
+      }
+    },
+    "uuts": {
+      "type": "array",
+      "description": "Collection of Unit Under Test (UUT) definitions",
+      "items": {
+        "$ref": "#/definitions/Uut"
+      }
+    },
+    "uutInstances": {
+      "type": "array",
+      "description": "Collection of specific UUT instances with serial numbers",
+      "items": {
+        "$ref": "#/definitions/UutInstance"
+      }
+    },
+    "testDescriptions": {
+      "type": "array",
+      "description": "Collection of test descriptions defining tests for specific UUTs",
+      "items": {
+        "$ref": "#/definitions/TestDescription"
+      }
+    }
+  },
+  "definitions": {
+    "Extensions": {
+      "type": "object",
+      "description": "Extension fields as string key-value pairs for custom metadata",
+      "additionalProperties": {
```

Comment:
Should this be an array?

Reply:
In order to use extension attributes, a user must define a schema and register it with the metadata store. How is the extension attribute schema defined and registered?